std/experimental/allocator/building_blocks/region.d

```diff
@@ -125,7 +125,7 @@ struct Region(ParentAllocator = NullAllocator,
     {
         static if (growDownwards)
```

Comment:
hoist the test and put it here right at the beginning: `if (n == 0) return null`